File: PROGRAMMERS/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    visited = {}
​
    for i, c in enumerate(s):
        if c in visited:
            diff = i - visited[c] + 1
            answer.append(diff)
            visited[c] = i + 1
        else:
            answer.append(-1)
            visited[c] = i + 1
​
​
​
    return answer

File: PROGRAMMERS/가장 가까운 같은 글자.java
import java.util.HashMap;
import java.util.Map;
​
class Solution {
    public int[] solution(String s) {
        int len = s.length();
        int[] answer = new int[len];
        Map<Character, Integer> map = new HashMap<>();
        
        for(int i = 0; i < len; i++) {
            char ch = s.charAt(i);
            answer[i] = i - map.getOrDefault(ch, i + 1);
            map.put(ch, i);
        }
        return answer;
    }
}